workflow/react-project/src/pages/SearchCocktail.js

```diff
@@ -0,0 +1,226 @@
+import './SearchCocktail.css';
+import React, { useState, useRef, useEffect } from 'react';
+import Modal from '../components/Modal';
+import { connect } from 'react-redux';
+import RandomCocktail from '../components/RandomCocktail';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCocktail, faSearch } from '@fortawesome/free-solid-svg-icons';
+import { showCocktailModal } from '../redux/action-creators/modalActions';
+import Loader from '../components/Loader';
+import { fetchSearchCocktails, cleanSearchResults } from '../redux/action-creators/searchCocktailActions';
+import _ from 'lodash';
+
+const SearchCocktail = (props) => {
+    const [search, setSearch] = useState('');
+    const [key, setKey] = useState(null);
+
+    const updateSearch = (event) => {
+        setSearch(event.target.value);
+    }
+
+    const debounceSearch = useRef(
+        _.debounce(search => {
+            props.fetchSearchCocktails(search);
+        }, 1000)
+    );
+
+    console.log(props.cocktails);
+    
+    useEffect(() => {
+        if (search) {
+            debounceSearch.current(addSearchParams());
+        }
+    },[search]);
+
+    useEffect(props.cleanSearchResults, []);
+
+    const setCardsAmount = (n) => {
+        let cards = [];
+        for(let i = 0; i < n; i++) {
+            const src = props.cocktails[i].strDrinkThumb;
+            const cocktailName = props.cocktails[i].strDrink;
+            let isAlco = false;
+            if (props.cocktails[i].strAlcoholic === 'Alcoholic') {
+                isAlco = true;
+            }
```

Comment:
`const isAlco = props.cocktails[i].strAlcoholic === 'Alcoholic'` can replace these 3 lines
also, use constants for strings like 'Alcoholic'